CMakeLists.txt

```
@@ -430,12 +434,12 @@ if(__INTEGRATE_OPENCL)
 endif()
 
 if(USE_CUDA)
-  set_target_properties(lightgbm PROPERTIES CUDA_RESOLVE_DEVICE_SYMBOLS ON)
+  set_target_properties(lightgbm PROPERTIES CUDA_RESOLVE_DEVICE_SYMBOLS ON CUDA_ARCHITECTURES ${CUDA_ARCH_VERSION})
   TARGET_LINK_LIBRARIES(
     lightgbm
     ${histograms}
   )
-  set_target_properties(_lightgbm PROPERTIES CUDA_RESOLVE_DEVICE_SYMBOLS ON)
+  set_target_properties(_lightgbm PROPERTIES CUDA_RESOLVE_DEVICE_SYMBOLS ON CUDA_ARCHITECTURES ${CUDA_ARCH_VERSION})
```

For better readability.
```suggestion
  set_target_properties(_lightgbm PROPERTIES CUDA_RESOLVE_DEVICE_SYMBOLS ON)
  set_target_properties(_lightgbm PROPERTIES CUDA_ARCHITECTURES ${CUDA_ARCH_VERSION})
```